Build fails with LTO

I tried to build with the following *FLAGS to optimize the build: `-flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing`

Note the -Werror=* flags are used to help detect cases where the compiler tries to optimize by assuming UB cannot exist in the source code -- if it does exist, ordinarily the code would be miscompiled, and this says to make the miscompilation a fatal error.

I got this error:
```
x86_64-pc-linux-gnu-gcc  -march=native -fstack-protector-all -O2 -pipe -fdiagnostics-color=always -frecord-gcc-switches -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-clash-protection -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing  -Wformat -Werror=format-security -Werror=implicit-function-declaration -Werror=implicit-int -Werror=int-conversion -Werror=incompatible-pointer-types  -Wl,-O1 -Wl,--as-needed -flto=4 -Werror=odr -Werror=lto-type-mismatch -Werror=strict-aliasing -Wl,--defsym=__gentoo_check_ldflags__=0 -o packETH src/pixmaps.o src/ui.o src/main.o src/support.o src/interface.o src/callbacks.o src/function.o src/function_send.o src/loadpacket.o src/savepacket.o  -lpthread -lm  -lgtk-3 -lgdk-3 -lharfbuzz -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo -lcairo-gobject -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lgthread-2.0 -Wl,--export-dynamic -lgmodule-2.0 -pthread -lglib-2.0 -lm
src/callbacks.c:63:22: error: type of ‘packet’ does not match original declaration [-Werror=lto-type-mismatch]
   63 | extern unsigned char packet[9300];
      |                      ^
src/function.c:55:15: note: array types have different bounds
   55 | unsigned char packet[100001];
      |               ^
src/function_send.c:50:22: error: type of ‘packet’ does not match original declaration [-Werror=lto-type-mismatch]
   50 | extern unsigned char packet[10000];
      |                      ^
src/function.c:55:15: note: array types have different bounds
   55 | unsigned char packet[100001];
      |               ^
src/function.c:55:15: note: ‘packet’ was previously declared here
lto1: some warnings being treated as errors
lto-wrapper: fatal error: x86_64-pc-linux-gnu-gcc returned 1 exit status
compilation terminated.
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:482: packETH] Error 1
```

Downstream report: https://bugs.gentoo.org/861695
Full build log: [build.log](https://github.com/jemcek/packETH/files/14629294/build.log)

--- a/src/callbacks.c
+++ b/src/callbacks.c
@@ -59,7 +59,6 @@ static	GtkWidget *tos_dialog_menu = NULL;
 static	GtkWidget *fragment_dialog_menu = NULL;
 static	GtkWidget *selection1_dialog = NULL;
 static	GtkWidget *udp_payload_dialog = NULL;
-extern unsigned char packet[9300];
 extern int number;
 extern int stop_flag;
 extern long desired_bw;

--- a/src/function.h
+++ b/src/function.h
@@ -23,6 +23,7 @@
 #include <gtk/gtk.h>
 
 extern char iftext[20];
+extern unsigned char packet[100001];
 
 signed int char2x(char *p);
 char c4(int value);

--- a/src/function_send.c
+++ b/src/function_send.c
@@ -47,7 +47,6 @@
 //#define 128 
 
 extern int number;
-extern unsigned char packet[10000];
 extern gboolean stop_flag;
 extern long li_sentbytes;
 extern long li_packets_sent;

--- a/src/savepacket.c
+++ b/src/savepacket.c
@@ -29,7 +29,6 @@
 #include "headers.h"
 
 extern int number;
-extern unsigned char packet[9300];
 
 /* Link-layer type; */
 static unsigned long pcap_link_type = 1;   /* Default is DLT-EN10MB */

